File: jlpt/native_android/app/src/main/java/com/jlpt/android/data/remote/dto/ExamDtos.kt
package com.jlpt.android.data.remote.dto

import com.jlpt.android.domain.model.AttemptStatus
import com.jlpt.android.domain.model.ExamType
import com.jlpt.android.domain.model.JlptLevel
import com.jlpt.android.domain.model.SkillType
import kotlinx.serialization.Serializable

@Serializable
data class ChoiceOptionResponseDto(
    val id: Long,
    val choiceText: String,
    val displayOrder: Int
)

@Serializable
data class QuestionForAttemptResponseDto(
    val id: Long,
    val skillType: SkillType,
    val questionText: String,
    val passageContent: String? = null,
    val listeningAudioId: Long? = null,
    val choices: List<ChoiceOptionResponseDto> = emptyList()
)

@Serializable
data class ExamSummaryResponseDto(
    val id: Long,
    val level: JlptLevel,
    val title: String,
    val examType: ExamType,
    val skillType: SkillType? = null,
    val timeLimitMinutes: Int,
    val questionCount: Int
)

@Serializable
data class ExamDetailResponseDto(
    val id: Long,
    val level: JlptLevel,
    val title: String,
    val examType: ExamType,
    val skillType: SkillType? = null,
    val timeLimitMinutes: Int,
    val questions: List<QuestionForAttemptResponseDto> = emptyList()
)

@Serializable
data class AttemptResponseDto(
    val id: Long,
    val examId: Long,
    val examTitle: String,
    val status: AttemptStatus,
    val startedAt: String,
    val submittedAt: String? = null,
    val score: Int? = null,
    val maxScore: Int? = null,
    val questions: List<QuestionForAttemptResponseDto> = emptyList()
)

@Serializable
data class SubmitAnswerRequestDto(
    val questionId: Long,
    val selectedChoiceId: Long? = null
)

@Serializable
data class AnswerResultResponseDto(
    val questionId: Long,
    val questionText: String,
    val selectedChoiceId: Long? = null,
    val correctChoiceId: Long,
    val correct: Boolean,
    val explanation: String? = null
)

@Serializable
data class AttemptResultResponseDto(
    val attemptId: Long,
    val examId: Long,
    val score: Int? = null,
    val maxScore: Int? = null,
    val submittedAt: String? = null,
    val answers: List<AnswerResultResponseDto> = emptyList()
)

File: jlpt/native_ios/Sources/Data/DTOs/ExamDTOs.swift
import Foundation

struct ChoiceOptionResponseDTO: Decodable {
    let id: Int64
    let choiceText: String
    let displayOrder: Int
}

struct QuestionForAttemptResponseDTO: Decodable {
    let id: Int64
    let skillType: SkillType
    let questionText: String
    let passageContent: String?
    let listeningAudioId: Int64?
    let choices: [ChoiceOptionResponseDTO]
}

struct ExamSummaryResponseDTO: Decodable {
    let id: Int64
    let level: JlptLevel
    let title: String
    let examType: ExamType
    let skillType: SkillType?
    let timeLimitMinutes: Int
    let questionCount: Int
}

struct ExamDetailResponseDTO: Decodable {
    let id: Int64
    let level: JlptLevel
    let title: String
    let examType: ExamType
    let skillType: SkillType?
    let timeLimitMinutes: Int
    let questions: [QuestionForAttemptResponseDTO]
}

struct AttemptResponseDTO: Decodable {
    let id: Int64
    let examId: Int64
    let examTitle: String
    let status: AttemptStatus
    let startedAt: String
    let submittedAt: String?
    let score: Int?
    let maxScore: Int?
    let questions: [QuestionForAttemptResponseDTO]
}

struct SubmitAnswerRequestDTO: Encodable {
    let questionId: Int64
    let selectedChoiceId: Int64?
}

struct AnswerResultResponseDTO: Decodable {
    let questionId: Int64
    let questionText: String
    let selectedChoiceId: Int64?
    let correctChoiceId: Int64
    let correct: Bool
    let explanation: String?
}

struct AttemptResultResponseDTO: Decodable {
    let attemptId: Int64
    let examId: Int64
    let score: Int?
    let maxScore: Int?
    let submittedAt: String?
    let answers: [AnswerResultResponseDTO]
}
